pkg/policy/client.go

```diff
@@ -162,7 +162,8 @@ func (c client) GetIngressBackendPolicy(svc service.MeshService) *policyV1alpha1
 		// Multiple IngressBackend policies for the same backend will be prevented
 		// using a validating webhook.
 		for _, backend := range ingressBackend.Spec.Backends {
-			if backend.Name == svc.Name {
+			// we need to check ports to allow ingress to multiple ports on the same svc
+			if backend.Name == svc.Name && backend.Port.Number == int(svc.TargetPort) && backend.Port.Protocol == svc.Protocol {
```

Comment:
Let's drop the protocol check here as it breaks existing behavior. The protocol on a service represents its protocol for east-west traffic (pod to pod), not north-south (external to pod). So a service with `http` as its internal mesh protocol can still use `https` as the protocol with ingress.